libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Memory/DialogStateManager.cs

```diff
@@ -123,14 +123,30 @@ string IMemory.Version()
         public virtual MemoryScope ResolveMemoryScope(string path, out string remainingPath)
         {
             var scope = path;
+            var sepIndex = -1;
             var dot = path.IndexOf(".");
-            if (dot > 0)
+            var openSquareBracket = path.IndexOf("[");
```

Comment:
>var openSquareBracket = path.IndexOf("[") [](start = 12, length = 41)

No description of what this change is about.
no unit tests testing the new functionality
Please add unit tests and comments.

Reply:
yes, I have added the detailed descriptions for this pr. 
But, I do now know where to put the unit test of DialogStateManager,  it doesn't seem to have unit tests originally.